common/kvstore/tablestore/table_store_builder.go

```diff
@@ -0,0 +1,381 @@
+package tablestore
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/common/kvstore"
+	"github.com/Layr-Labs/eigenda/common/kvstore/leveldb"
+	"github.com/Layr-Labs/eigenda/common/kvstore/mapstore"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"math"
+	"sort"
+)
+
+// The table ID reserved for the metadata table. The metadata table is used for internal bookkeeping.
+// The following data is currently stored in the metadata table:
+//   - Schema version (in case we ever need to do a schema migration)
+//   - Table deletion markers used to detect a crash during table deletion and to complete
```

Comment:
If you adopt lazy deletion, then this can be a separate DeletionTable, which records each deleted table as an entry in this table. The background thread work on the DeletionTable and perform GC.